src/Databases/MySQL/MaterializeMySQLSyncThread.cpp

```diff
@@ -326,9 +329,10 @@ static inline void dumpDataForTables(
             tryToExecuteQuery(query_prefix + " " + iterator->second, query_context, database_name, comment); /// create table.
 
             auto out = std::make_shared<CountingBlockOutputStream>(getTableOutput(database_name, table_name, query_context));
+            StreamSettings ss(context->getSettingsRef());
```

Comment:
This is only dump tables (and some other additional info fetches), which is done through `MySQLBlockInputStream`, but synchronization is done here https://github.com/ClickHouse/ClickHouse/blob/master/src/Databases/MySQL/MaterializeMySQLSyncThread.cpp#L785. Then your settings will not be used.

Reply:
BTW, there are already such settings: https://github.com/ClickHouse/ClickHouse/blob/master/src/Databases/MySQL/MaterializeMySQLSettings.h#L12-L13

Reply:
No, In this pr the settings are controlled the data sync when select * from MySQL. Contains:
1. MySQL table engine
2. MySQL dictionary
3. MaterializeMySQL

And in MaterializeMySQL database settings, it just controls the data sync when consuming MySQL Binlog. 

Maybe my pr description is not very accurate.

Reply:
Ok, lets write it into changelog entry, because now it refers only to MaterializeMySQL.

Reply:
> This only dumps tables (and some other additional info fetches), which is done through MySQLBlockInputStream, but synchronization is done here https://github.com/ClickHouse/ClickHouse/blob/master/src/Databases/MySQL/MaterializeMySQLSyncThread.cpp#L785. Then your settings will not be used.


Oh, the MaterializeMySQL  just has two main steps:
1. sync history data
2. consume new data.

In `prepareSynchronized`, it will SELECT * FROM MYSQL's full data. And every block has at most 65505 rows. 

In `onevent`, it consumes MySQL binlog.

And in some cases: people use MySQL table with a large size in one row. So that in some low memory instances, 

if the table rows are larger than 65505 then, in `step 1 sync history data` we need a block contains 65505 rows data. The size in memory may be more than the instance's total memory. 

This will let ClickHouse be OOM.

I think because my pr description is not very accurate. Let you have some mistakes.

Reply:
> Ok, lets write it into changelog entry, because now it refers only to MaterializeMySQL.

Ok, I will do it later.